src/clj_ldap/client.clj

```diff
@@ -99,10 +100,11 @@
 
 (defn- connection-options
   "Returns a LDAPConnectionOptions object"
-  [{:keys [connect-timeout timeout]}]
+  [{:keys [connect-timeout timeout verify-host?]}]
   (let [opt (LDAPConnectionOptions.)]
     (when connect-timeout (.setConnectTimeoutMillis opt connect-timeout))
     (when timeout         (.setResponseTimeoutMillis opt timeout))
+    (when verify-host?    (.setSSLSocketVerifier opt (HostNameSSLSocketVerifier. false)))
```

Comment:
The `false` flag on `HostNameSSLSocketVerifier` is for `allowWidlcards`. I wasn't sure whether to permit this. Any comments?

Reply:
Personally I would say we should use true the allowWildCards as if a customer has set a wildcard in their cert then it could be legitimate but could fail.

An alternative would be to make this parameter configurable but seems like a lot of work for not that much gain.

Reply:
My instinct is to make this an option that can be specified, and default to false. @jsane and @joshcooper have any thoughts around this?

Reply:
While there seems to be widespread industry practice of using wildcard certificates for a variety of reasons it is not endorsed or mentioned in any pkix specs. On the contrary rfc6125 discourages use of wildcards. The problem with wildcards is the matching rules are not standard and varied. It's possible that some of our customers might actually be using them and disallowing wildcards w/o any configuration option is likely to break things. 
But allowing them seems an insecure choice. I imagine (& know) it would be much easier for an insider to obtain a certificate within an organization that will match. That defeats the purpose of what this bug is trying to address. 
So an additional option to allow/disallow wildcards (default to false) would be preferable so whoever can make a conscious choice. 

If we decide to always allow wildcards then we may need to document the behavior of our wildcard matching (which could be a time consuming exercise) and provide a security warning of potential implications with it.